src/routes/communityGroups/community_groups.contoller.js

```diff
@@ -0,0 +1,82 @@
+const fs = require("fs");
+const { promisify } = require("util");
+
+const { ERROR_CODES } = require("../../middleware/error-handler");
+
+const data = require("../../../db/communityGtoups.data.json");
+
+const writeFile = promisify(fs.writeFile);
+
+const dbPath = "db/communityGtoups.data.json";
+
+// return all community groups
+const listCommunityGroups = (req, res) => {
+  return res.status(200).json(data);
+};
+
+// get a community group by its id if it exists
+const getCommunityGroup = (req, res) => {
+  const record = data.filter(
```

Comment:
How might you look up a record by it's id more quickly than scanning an array? (hint: think data structures)

Reply:
Similar to another comment, the immediate solution would be restructuring the data object to have a structure like so:
```
{
  "0": {
    "name": "Murphy - Wyman",
    "description": "A group that cares about Cheese",
    "location": "Andersonport, Qatar"
  }
}
```

The structure used was the one provided to us though (and only on a temporary basis till it's replaced with DB integration), so I didn't put a lot of thought into restructuring it.
Not sure if you had a different idea in mind. I'd love to discuss alternatives!

Reply:
Nope, that's exactly what I was going for. good job!